discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py

```diff
@@ -211,11 +210,33 @@ def get_primary_machine_cidr(self):
         return self._config.net_asset.machine_cidr
 
     def get_provisioning_cidr(self):
-        # In dualstack/IPv6 mode the secondary network is IPv6
-        if self.is_ipv6:
+        # In dualstack mode the primary network is IPv4
+        if self.is_ipv6 and not self.is_ipv4:
             return self._config.net_asset.provisioning_cidr6
         return self._config.net_asset.provisioning_cidr
 
+    def get_all_machine_addresses(self):
+        addresses = []
+
+        if self.is_ipv4:
+            addresses.append(self._config.net_asset.machine_cidr)
+
+        if self.is_ipv6:
+            addresses.append(self._config.net_asset.machine_cidr6)
+
+        return addresses
+
+    def get_all_provisioning_addresses(self):
```

Comment:
As above, future us will be thankful for a comment that it's not "get all" but "get one of each"

Reply:
sure, done